src/tests/Grape/TraceGenerators/EventPipeTraceGenerator.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Diagnostics.Tracing;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.IO;
+using Microsoft.Diagnostics.NETCore.Client;
+using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Parsers;
+using Microsoft.Diagnostics.Tracing.Parsers.Kernel;
+
+
+
+namespace Microsoft.Diagnostics.Grape
+{
+	public class EventPipeTraceGenerator
+	{
+		TestRunner _runner;
+		string _pathToExe;
+		string _traceName;
+		List<EventPipeProvider> _providers;
+
+		public EventPipeTraceGenerator(string pathToExe, string traceName, List<EventPipeProvider> providers)
+		{
+			_pathToExe = pathToExe;
+			_traceName = traceName;
+			_providers = providers;
+		}
+
+	    public void Collect(int duration)
+	    {
+	    	var pid = LaunchProcess(_pathToExe);
+	    	TraceProcessForDuration(pid, duration, _traceName);
+	    }
+
+		private int LaunchProcess(string pathToExe)
+	    {
+	        _runner = new TestRunner(pathToExe);
+	        // Sleep for some time until diagnostics server pipe gets created
```

Comment:
Could we poll for the pipe instead?  We should know the pid by this point.  I guess this won't matter once we have startup tracing, though.